### ISSUE TITLE
PullRequest-Sanity-JDK11 JCL_Test_OutOfMemoryError_1 ASSERTION FAILED ** at vmthread.c:1432: ((fatalRecursiveStackOverflowDetected))

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#2902]
This problem occurred in Pull Request testing on a change to use Java 11 bytecodes (javac target 11) when doing the Java 11 builds. https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/21

https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK11-linux_390-64_cmprssptrs-OpenJDK11/3/tapResults/

```
===============================================
Running test JCL_Test_OutOfMemoryError_1 ...
===============================================
JCL_Test_OutOfMemoryError_1 Start Time: Mon Sep 17 15:56:25 2018 Epoch Time (ms): 1537214185082
test with -XX:RecreateClassfileOnload
JVMJ9VM019E Unrecoverable error: Unable to find and initialize required class Ljava/lang/Object;Ljava/util/Comparator<Ljava/lang/String;>;Ljava/io/Serializable;
JVMJ9VM023I This may indicate that JAVA_HOME is incorrect, or that class libraries are not installed
19:56:25.270 0x14f00    j9vm.224    *   ** ASSERTION FAILED ** at vmthread.c:1432: ((fatalRecursiveStackOverflowDetected))

JCL_Test_OutOfMemoryError_1_FAILED
```

**Comment by JasonFengJ9** [comment]
Note: @fengxue-IS is on this.

**Comment by DanHeidinga** [comment]
Possibly another occurrence of https://github.com/eclipse/openj9/issues/2708

**Comment by DanHeidinga** [comment]
The build in ibmruntimes/openj9-openjdk-jdk11#21 passes with the fix from #2708.  Closing this.
